inc/shortcodes/class-iframe.php

```diff
@@ -49,6 +49,29 @@ public static function get_whitelisted_iframe_domains() {
 		return apply_filters( 'shortcake_bakery_whitelisted_iframe_domains', array() );
 	}
 
+	/**
+	 * Transform any <iframe> embeds within content to our iframe shortcode
+	 *
+	 * @param string $content
+	 * @return string
+	 */
+	public static function reversal( $content ) {
+
+		if ( $iframes = self::parse_iframes( $content ) ) {
+			$whitelisted_iframe_domains = static::get_whitelisted_iframe_domains();
+			$replacements = array();
+			foreach ( $iframes as $iframe ) {
+				if ( ! in_array( parse_url( $iframe->src_force_protocol, PHP_URL_HOST ), $whitelisted_iframe_domains ) ) {
+					continue;
+				}
+				$replacements[ $iframe->original ] = '[' . self::get_shortcode_tag() . ' src="' . esc_url_raw( $iframe->attrs['src'] ) . '"]';
```

Comment:
@goldenapples Do you think there's rationale for persisting iframe attributes too? If so, which, and how should we go about it?